torchgeo/datasets/ssl4eo_benchmark_landsat.py

```diff
@@ -0,0 +1,348 @@
+            self.root,
+            self.mask_dir_name,
+            "**",
+            f"{self.mask_product}_{self.year_dict[self.input_sensor]}.tif",
+        )
+        for fname in glob.iglob(mask_pathname, recursive=True):
+            exists.append(True)
+            break
+        if exists and all(exists):
+            return
+
+        # Check if the tar.gz files have already been downloaded
+        exists = []
+        img_pathname = os.path.join(self.root, f"{self.img_dir_name}.tar.gz")
+        if os.path.exists(img_pathname):
+            exists.append(True)
+        mask_pathname = os.path.join(self.root, f"{self.mask_dir_name}.tar.gz")
+        if os.path.exists(mask_pathname):
+            exists.append(True)
+        if exists and all(exists):
+            self._extract()
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        # download imagery
+        download_url(
+            self.url.format(self.img_dir_name, self.img_dir_name),
+            self.root,
+            md5=self.img_md5s[self.input_sensor] if self.checksum else None,
+        )
+        # download mask
+        download_url(
+            self.url.format(self.mask_dir_name, self.mask_dir_name),
+            self.root,
+            md5=self.mask_md5s[self.input_sensor.split("_")[0]][self.mask_product]
+            if self.checksum
+            else None,
+        )
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        img_pathname = os.path.join(self.root, f"{self.img_dir_name}.tar.gz")
+        extract_archive(img_pathname)
+
+        mask_pathname = os.path.join(self.root, f"{self.mask_dir_name}.tar.gz")
+        extract_archive(mask_pathname)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            image and sample
+        """
+        img_path, mask_path = self.sample_collection[index]
+
+        sample = {
+            "image": self._load_image(img_path),
+            "mask": self._load_mask(mask_path),
+        }
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.sample_collection)
+
+    def retrieve_sample_collection(self) -> list[tuple[str]]:
```

Comment:
```suggestion
    def retrieve_sample_collection(self) -> list[tuple[str, str]]:
```
This will fix most other mypy errors